src/bridge/MemberListSyncer.ts

```diff
@@ -278,19 +277,28 @@ export class MemberListSyncer {
         // map the filtered rooms to a list of users to join
         // [Room:{reals:[uid,uid]}, ...] => [{uid,roomid}, ...]
         const entries: { roomId: string; displayName: string; userId: string; frontier: boolean}[] = [];
-        filteredRooms.forEach((roomInfo) => {
-            roomInfo.realJoinedUsers.forEach((uid, index) => {
+        const idleRegex = this.server.ignoreIdleUsersOnStartupExcludeRegex;
+        for (const roomInfo of filteredRooms) {
+            for (const uid of roomInfo.realJoinedUsers) {
+                if (this.server.ignoreIdleUsersOnStartup) {
+                    const idle = await this.ircBridge.activityTracker?.isUserOnline(
+                        uid, this.server.ignoreIdleUsersOnStartupAfterMs, false
+                    );
+                    if (!(idle?.online) && !idleRegex?.exec(uid)) {
+                        continue;
```

Comment:
The effect will be some extra users on the matrix side which are not shown on irc side? If the user on matrix side becomes active (ie writes a message, for example), do things just work and the user is joined to the irc side?

Reply:
If the user sends a message then yeah they will be joined/connected. A tiny proportion of users who did an action while the bridge is down may be missed.